File: pseudo-palindromic paths in a binary tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    HashMap<Integer,Integer>mp=new HashMap<Integer,Integer>();
    int res=0;
    boolean isLeaf(TreeNode root){
        if(root.left==null&&root.right==null){return true;}
        return false;
    }
    void ok(){
        int sz=0,odd=0,even=0;
        for(Integer i:mp.keySet()){
            int key=i,value=mp.get(i);
            if(value>0){
                sz+=value;
                if(value%2==1){odd+=1;}
                else{even+=1;}
            }
        }
        if(sz%2==1){if(odd==1){res++;}}
        else{if(odd==0){res++;}}
    }
    void dfs(TreeNode root){
        if(root==null){return;}
        if(mp.containsKey(root.val)){
            mp.put(root.val,mp.get(root.val)+1);
        }
        else{mp.put(root.val,1);}
        if(isLeaf(root)){
            ok();
        }
        if(root.left!=null){dfs(root.left);}
        if(root.right!=null){dfs(root.right);}
        mp.put(root.val,mp.get(root.val)-1);
        return;
    }
    public int pseudoPalindromicPaths (TreeNode root) {
        dfs(root);
        return res;
    }
}

File: pseudo-palindromic paths in a binary tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
unordered_map<int,int>mp;
int res=0;
void ok(){
    int sz=0,odd=0,even=0;
   // cout<<"\n";
    for(pair<int,int>p:mp){
        if(p.second){
        //    cout<<"val:"<<p.first<<",count:"<<p.second;
            cout<<'\n';
            (p.second&1)?odd++:even++;
            sz+=p.second;
        }
    }
   // cout<<"\n";
    if(sz&1){if(odd==1){res++;}}
    else{if(odd==0){res++;}}
    return;
}
bool isleaf(TreeNode* root){
   if(!root->left&&!root->right){return true;}
   return false;
}
void dfs(TreeNode* root){
    if(root==nullptr){return ;}
    mp[root->val]++;
    if(isleaf(root)){
        ok();
    }
    if(root->left){dfs(root->left);}
    if(root->right){dfs(root->right);}
    mp[root->val]--;
}
    int pseudoPalindromicPaths (TreeNode* root) {
        dfs(root);
        return res;
    }
};